finetune/nn/target_blocks.py

```diff
@@ -23,16 +21,15 @@ def perceptron(x, ny, config, w_init=None, b_init=None):
     """
     w_init = w_init or tf.random_normal_initializer(stddev=config.weight_stddev)
     b_init = b_init or tf.constant_initializer(0)
-    with tf.variable_scope("perceptron"):
-        nx = config.n_embed
-        if config.use_auxiliary_info:
-            nx += config.n_context_embed
+
+    with tf.variable_scope('perceptron'):
+        nx = shape_list(x)[-1]
         w = tf.get_variable("w", [nx, ny], initializer=w_init)
         b = tf.get_variable("b", [ny], initializer=b_init)
         return tf.matmul(x, w) + b
 
 
-def language_model(*, X, M, embed_weights, hidden, config, reuse=None):
+def language_model(*, X, M, embed_weights, hidden, config, reuse=None, train=False):
```

Comment:
its a bit of syntax to stop you passing anything as positional arguments. I like it a lot for tensorflow stuff because you can accidentally swap stuff round and not necessarily notice. Eg. X and M here could be directly swapped.